Minor tidyup

aeijdenberg opened the pull request:
Might be subjective, but 2 small tidy up suggestions...

Make internal mutex private, and make signature of public LoaderOpt noot use private class

aeijdenberg committed:
Make internal mutex private, and make signature of public LoaderOpt not use private class

jonathaningram commented on getter.go:
```diff
@@ -11,8 +11,8 @@ type Key string
-	m            map[Key]*Template
-	sync.RWMutex // RWMutex protects m.
+	m    map[Key]*Template
+	lock sync.RWMutex // RWMutex protects m.
```
Thanks @aeijdenberg ! Can you please rename `lock` to `mu` and change comment to `mu protects m.`?

Former removes stutter too: `ts.lock.RLock()` --> `ts.mu.RLock()`.

Aside: I just see now after my rename I forgot to rename the receiver to `g` (it's still `ts).

jonathaningram commented on load.go:
```diff
@@ -17,17 +17,17 @@ const (
 
-// loader is a type which loads email templates located at rootPath.
-type loader struct {
+// Loader is a type which loads email templates located at rootPath.
```
@aeijdenberg interesting. Yeah I didn't want to export `Loader` at all (it's not needed, low API surface footprint). So even though `LoaderOpt` refers to unexported type: `type LoaderOpt func(l *loader)` is it a problem because callers will only ever be referring to `LoaderOpt` anyway. Thoughts?

aeijdenberg replied:
It saw it when reading the `godoc` - it's unusual to see a private class referenced in `godoc`, so I figured it was probably unusual style.

jonathaningram replied:
Cool. So I was just wondering if there was a way to do this without exporting `Loader` since it's an empty struct with no methods as far as callers can see. But then I realised that there may actually be times that we want to allow the caller to set stuff on the loader. We'll leave the fields unexported for now, but imagine something like this:

```go
// WithCustomerRootPath pulls the customer template path from the request form.
// Maybe it would accept a customer struct or just a string to the path directly, this is just an example.
func WithCustomerRootPath(rootPath string, r *http.Request) emailtemplate.LoaderOpt {
	return func(l *emailtemplate.Loader) {
		l.RootPath = filepath.Join(rootPath, r.FormValue("customerTemplatePath")),
	}
}

func (w http.ResponseWriter, r *http.Request) {
	// Maybe not the best idea to load in a request anyway...
	g, err := Load(
		WithCustomerRootPath("./templates", r),
	)
}
```

jonathaningram replied:
tl;dr ^ - we may as export `Loader` as you've done.

aeijdenberg committed:
Rename per feedback.

Use defer() for unlocks

Remove nil check for len() whose only purpose appears to have been to appease a test, which shouldn't be relying on the object value returned if error is set anyway.

jonathaningram commented on load_test.go:
```diff
@@ -97,9 +97,9 @@ func TestLoaderLoad(t *testing.T) {
-			}
-			if g.len() != tt.found {
-				t.Errorf("load(): getter.len() = %d, want %d", g.len(), tt.found)
+				if g.len() != tt.found {
```
@aeijdenberg thanks, you nailed it.
